### PR TITLE
coreos-base/oem-ec2-compat: use ttyS0 for console

### PR DESCRIPTION
**Pull request by crawford**
On HVM instances, tty0 is actually a graphical interface. Since the
interface exposed is non-interactive, it's not possible to scroll
through the logs, which results in lots of missing information. Setting
the console to ttyS0, on the other hand, sends the output to the system
log.

PV instances don't have this issue since they are xen-based and use
hvc0.

### CODE REVIEW

File: coreos-base/oem-ec2-compat/files/grub-ec2.cfg [hunk 1]
```diff
@@ -5,3 +5,7 @@ set oem_id="ec2"
 # Blacklist the Xen framebuffer module so it doesn't get loaded at boot
 # Disable `ens3` style names, so eth0 is used for both ixgbevf or xen.
 set linux_append="modprobe.blacklist=xen_fbfront net.ifnames=0"
+
+if [ "$grub_platform" = pc ]; then
+	set linux_console="console=ttyS0,115200n8"
```
Comment (marineam) [inline coreos-base/oem-ec2-compat/files/grub-ec2.cfg]: Add grub serial so you see it's output too:

          serial com0 --speed=115200 --word=8 --parity=no
￼          terminal_input console serial_com0
￼          terminal_output console serial_com0

Comment (crawford) [inline coreos-base/oem-ec2-compat/files/grub-ec2.cfg]: That's an unfortunate side-effect. Added.

File: coreos-base/oem-ec2-compat/files/grub-ec2.cfg [hunk 2]
```diff
@@ -5,3 +5,10 @@ set oem_id="ec2"
...
```
Comment (marineam) [inline coreos-base/oem-ec2-compat/files/grub-ec2.cfg]: Very intentional, if the config has customized the linux console it is responsible for configuring grub too.

Oh, but you can actually drop console, so grub also doesn't do the dual output, only writing to serial_com0

Comment (marineam) [inline coreos-base/oem-ec2-compat/files/grub-ec2.cfg]: So forcing you to configure grub actually makes grub behave correctly too! :-P

**Pull request merged**